internal/tofu/eval_variable.go

```diff
@@ -435,6 +442,9 @@ You can correct this by removing references to sensitive values, or by carefully
 				})
 				errorMessage = "The error message included a sensitive value, so it will not be displayed."
 			} else {
+				// errorValue could have some marks other than sensitive ones,
+				// so we need to unmark to not get panic from AsString()
+				errorValue, _ = errorValue.UnmarkDeep()
```

Comment:
Let's swap this to Unmark() for safety.  Given that the next call is AsString(), it should be sufficient.

Reply:
Changed it to remove only deprecated marks for the same reason as in the previous comment: [7ada323](https://github.com/opentofu/opentofu/pull/2633/commits/7ada323bce62fa7fc55a87b402a1447eacd4fb38)